multiformats_cid/cid.py

```diff
@@ -88,16 +99,16 @@ def buffer(self):
         """
         return self.multihash
 
-    def encode(self):
+    def encode(self) -> bytes:
         """
         base58-encoded buffer
 
         :return: encoded representation or CID
         :rtype: bytes
         """
-        return ensure_bytes(base58.b58encode(self.buffer))
+        return bytes(ensure_bytes(base58.b58encode(self.buffer)))
```

Comment:
There are a couple places throughout these changes where you will notice what appear to be arbitrary casts of things like this one. This is because these are method calls to untyped dependencies (which we are -- currently ignoring). A similar pattern with the same explanation is when you see something like:
```py
return f(x)
```
being replaced by

```py
results: bytes f(x)
return results
```